packages/react-components/src/echarts/extensions/minMaxDataStreamValues/updateDataStreamMinMax.ts

```diff
@@ -0,0 +1,20 @@
+import LineSeriesModel from 'echarts/types/src/chart/line/LineSeries';
+import { EChartsExtensionInstallRegisters } from 'echarts/types/src/extension';
+import { findMinMax, handleMinMaxSync } from './minMaxDataStreamSync';
+
+// Echarts core use type does not map correctly to the echarts extension type so exporting as any
+// eslint-disable-next-line
+export const dataStreamMinMaxSyncExtension: any = (registers: EChartsExtensionInstallRegisters) => {
+  registers.registerUpdateLifecycle(
+    'series:afterupdate',
+    (_ecModel, _api, params) => {
+      if (!params.updatedSeries) return;
+
+      params.updatedSeries.forEach((series) => {
+        const { max, min } = findMinMax(series);
+        if (max == null || min == null) return;
+        handleMinMaxSync(series as unknown as LineSeriesModel, max, min);
```

Comment:
can we write this as a function which does not depend on echarts values? seems like it just needs the id and the values?